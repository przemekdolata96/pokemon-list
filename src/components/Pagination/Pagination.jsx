import './Pagination.scss';
import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  currentPageNumber: PropTypes.number,
  lastPageNumber: PropTypes.number,
  last: PropTypes.func.isRequired,
};

const defaultProps = {
  currentPageNumber: 1,
  lastPageNumber: 9,
};

const Pagination = (props) => {
  const paginationItems = [];

  for (let i = 1; i <= props.lastPageNumber; i++) {
    paginationItems.push(
      <div
        key={i}
        className={props.currentPageNumber === i ? 'active' : ''}
        onClick={() => props.last(i)}
        onKeyDown={() => props.last(i)}
        role="button"
        tabIndex={i}
      >
        {i}
      </div>,
    );
  }

  return (
    <div>
      <div className="pagination-container">
        {paginationItems}
      </div>
    </div>
  );
};

Pagination.propTypes = propTypes;
Pagination.defaultProps = defaultProps;
export default Pagination;
