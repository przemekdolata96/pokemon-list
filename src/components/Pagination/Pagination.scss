@import '../../theme/variables.scss';

.pagination-container {
  display: flex;
  
  div {
    padding: 10px 10px;
    border: 0.5px solid $gray;
    border-right: none;
    cursor: pointer;

    &:last-of-type {
      border-right: 0.5px solid $gray;
    }
  }

  .active {
    color: $blue;
  }
}