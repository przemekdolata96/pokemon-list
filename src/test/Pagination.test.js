import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Pagination from '../components/Pagination/Pagination';

Enzyme.configure({
  adapter: new Adapter(),
});

test('renders without crashing', () => {
  const pagination = shallow(
    <Pagination
      lastPageNumber={8}
      currentPageNumber={1}
      maxPages={8}
      last={() => {}}
    />,
  );
  expect(pagination).toMatchSnapshot();
});
