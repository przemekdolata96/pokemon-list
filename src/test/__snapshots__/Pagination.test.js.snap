// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination
    currentPageNumber={1}
    last={[Function]}
    lastPageNumber={8}
    maxPages={8}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="pagination-container"
      >
        <div
          className="active"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={1}
        >
          1
        </div>
        <div
          className=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={2}
        >
          2
        </div>
        <div
          className=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={3}
        >
          3
        </div>
        <div
          className=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={4}
        >
          4
        </div>
        <div
          className=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={5}
        >
          5
        </div>
        <div
          className=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={6}
        >
          6
        </div>
        <div
          className=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={7}
        >
          7
        </div>
        <div
          className=""
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={8}
        >
          8
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="active"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={1}
          >
            1
          </div>,
          <div
            className=""
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={2}
          >
            2
          </div>,
          <div
            className=""
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={3}
          >
            3
          </div>,
          <div
            className=""
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={4}
          >
            4
          </div>,
          <div
            className=""
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={5}
          >
            5
          </div>,
          <div
            className=""
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={6}
          >
            6
          </div>,
          <div
            className=""
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={7}
          >
            7
          </div>,
          <div
            className=""
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={8}
          >
            8
          </div>,
        ],
        "className": "pagination-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": 1,
            "className": "active",
            "onClick": [Function],
            "onKeyDown": [Function],
            "role": "button",
            "tabIndex": 1,
          },
          "ref": null,
          "rendered": 1,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": 2,
            "className": "",
            "onClick": [Function],
            "onKeyDown": [Function],
            "role": "button",
            "tabIndex": 2,
          },
          "ref": null,
          "rendered": 2,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": 3,
            "className": "",
            "onClick": [Function],
            "onKeyDown": [Function],
            "role": "button",
            "tabIndex": 3,
          },
          "ref": null,
          "rendered": 3,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": 4,
            "className": "",
            "onClick": [Function],
            "onKeyDown": [Function],
            "role": "button",
            "tabIndex": 4,
          },
          "ref": null,
          "rendered": 4,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "children": 5,
            "className": "",
            "onClick": [Function],
            "onKeyDown": [Function],
            "role": "button",
            "tabIndex": 5,
          },
          "ref": null,
          "rendered": 5,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "host",
          "props": Object {
            "children": 6,
            "className": "",
            "onClick": [Function],
            "onKeyDown": [Function],
            "role": "button",
            "tabIndex": 6,
          },
          "ref": null,
          "rendered": 6,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "host",
          "props": Object {
            "children": 7,
            "className": "",
            "onClick": [Function],
            "onKeyDown": [Function],
            "role": "button",
            "tabIndex": 7,
          },
          "ref": null,
          "rendered": 7,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "8",
          "nodeType": "host",
          "props": Object {
            "children": 8,
            "className": "",
            "onClick": [Function],
            "onKeyDown": [Function],
            "role": "button",
            "tabIndex": 8,
          },
          "ref": null,
          "rendered": 8,
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="pagination-container"
        >
          <div
            className="active"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={1}
          >
            1
          </div>
          <div
            className=""
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={2}
          >
            2
          </div>
          <div
            className=""
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={3}
          >
            3
          </div>
          <div
            className=""
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={4}
          >
            4
          </div>
          <div
            className=""
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={5}
          >
            5
          </div>
          <div
            className=""
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={6}
          >
            6
          </div>
          <div
            className=""
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={7}
          >
            7
          </div>
          <div
            className=""
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={8}
          >
            8
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="active"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={1}
            >
              1
            </div>,
            <div
              className=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={2}
            >
              2
            </div>,
            <div
              className=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={3}
            >
              3
            </div>,
            <div
              className=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={4}
            >
              4
            </div>,
            <div
              className=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={5}
            >
              5
            </div>,
            <div
              className=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={6}
            >
              6
            </div>,
            <div
              className=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={7}
            >
              7
            </div>,
            <div
              className=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={8}
            >
              8
            </div>,
          ],
          "className": "pagination-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": 1,
              "className": "active",
              "onClick": [Function],
              "onKeyDown": [Function],
              "role": "button",
              "tabIndex": 1,
            },
            "ref": null,
            "rendered": 1,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": 2,
              "className": "",
              "onClick": [Function],
              "onKeyDown": [Function],
              "role": "button",
              "tabIndex": 2,
            },
            "ref": null,
            "rendered": 2,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": 3,
              "className": "",
              "onClick": [Function],
              "onKeyDown": [Function],
              "role": "button",
              "tabIndex": 3,
            },
            "ref": null,
            "rendered": 3,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "host",
            "props": Object {
              "children": 4,
              "className": "",
              "onClick": [Function],
              "onKeyDown": [Function],
              "role": "button",
              "tabIndex": 4,
            },
            "ref": null,
            "rendered": 4,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "host",
            "props": Object {
              "children": 5,
              "className": "",
              "onClick": [Function],
              "onKeyDown": [Function],
              "role": "button",
              "tabIndex": 5,
            },
            "ref": null,
            "rendered": 5,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "6",
            "nodeType": "host",
            "props": Object {
              "children": 6,
              "className": "",
              "onClick": [Function],
              "onKeyDown": [Function],
              "role": "button",
              "tabIndex": 6,
            },
            "ref": null,
            "rendered": 6,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "7",
            "nodeType": "host",
            "props": Object {
              "children": 7,
              "className": "",
              "onClick": [Function],
              "onKeyDown": [Function],
              "role": "button",
              "tabIndex": 7,
            },
            "ref": null,
            "rendered": 7,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "8",
            "nodeType": "host",
            "props": Object {
              "children": 8,
              "className": "",
              "onClick": [Function],
              "onKeyDown": [Function],
              "role": "button",
              "tabIndex": 8,
            },
            "ref": null,
            "rendered": 8,
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
